letsencrypt/plugins/common.py

```diff
@@ -136,7 +155,41 @@ def get_port(self):
 
     def get_addr_obj(self, port):
         """Return new address object with same addr and new port."""
-        return self.__class__((self.tup[0], port))
+        return self.__class__((self.tup[0], port), self.ipv6)
+
+    def _normalize_ipv6(self, addr):
+        """Return IPv6 address in normalized form, helper function"""
+        addr = addr.lstrip("[")
+        addr = addr.rstrip("]")
+        return self._explode_ipv6(addr)
+
+    def get_ipv6_exploded(self):
+        """Return IPv6 in normalized form"""
+        if self.ipv6:
+            return ":".join(self._normalize_ipv6(self.tup[0]))
+        return ""
+
+    def _explode_ipv6(self, addr):
+        """Explode IPv6 address for comparison"""
+        result = ['0', '0', '0', '0', '0', '0', '0', '0']
+        addr_list = addr.split(":")
+        append_to_end = False
+        for i in range(0, len(addr_list)):
+            block = addr_list[i]
+            if len(block) == 0:
+                # encountered ::, so rest of the blocks should be
+                # appended to the end
+                append_to_end = True
+                continue
+            elif len(block) > 1:
+                # remove leading zeros
+                block = block.lstrip("0")
+            if not append_to_end:
+                result[i] = str(block)
+            else:
+                # count the location from the end using negative indices
+                result[i-len(addr_list)] = str(block)
```

Comment:
It is correct, however I'm not currently checking if len(addr_list) <= len(result) in this method which - if false, would lead to IndexError. Going to add that check.